patroni/postgresql/__init__.py

```diff
@@ -645,12 +631,10 @@ def controldata(self):
 
     @contextmanager
     def get_replication_connection_cursor(self, host='localhost', port=5432, database=None, **kwargs):
-        replication = self.config.replication
-        extra_kwargs = {k: v for k, v in replication.items()
-                        if k not in ('username', 'password', 'connect_timeout', 'options')}
-        with get_connection_cursor(host=host, port=int(port), database=database or self._database, replication=1,
-                                   user=replication['username'], password=replication.get('password'),
-                                   connect_timeout=3, options='-c statement_timeout=2000', **extra_kwargs) as cur:
+        conn_kwargs = self.config.replication.copy()
+        conn_kwargs.update(host=host, port=int(port), database=database or self._database, connect_timeout=3,
+                           user=conn_kwargs.pop('username', None), replication=1, options='-c statement_timeout=2000')
```

Comment:
Looks like before lack of `username` key will cause an exception, which is not the case now. Is it intentional?

```diff
-user=replication['username']
+user=conn_kwargs.pop('username', None)
```

Reply:
Replication user name is mandatory, so the answer is no.
It should be just `conn_kwargs.pop('username')`.